static/js/subs.js

```diff
@@ -742,24 +741,34 @@ $(function () {
             return s.trim();
         });
 
-        for(i=0; i<users.length;i++)
-        {
-            var names = users[i].full_name.toLowerCase().split(/\s+/);
+        var filtered_users = _.filter(users, function (user) {
+            var person = people.get_by_email(user.email);
+            if (!person || !person.full_name) {
+                return false;
+            }
+            var names = person.full_name.toLowerCase().split(/\s+/);
             names = _.map(names, function (s) {
                 return s.trim();
             });
-
-            _.any(search_terms, function (search_term) {
-                $("label[for='" + users[i].email + "']").css({"display":"none"});
+            return _.any(search_terms, function (search_term) {
                 return _.any(names, function (name) {
-                    if (name.indexOf(search_term) === 0) {
-                        $("label[for='" + users[i].email + "']").css({"display":"inline-block"});
-                        return;
-                    }
-
+                    return name.indexOf(search_term) === 0;
                 });
             });
-        }
+        });
+
+        // Filter users map
```

Comment:
You may as well just generate the `filtered_user_map` directly. No need to create a `filtered_users` list first